cmd/snap/cmd_run.go

```diff
@@ -979,3 +1017,132 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 		return syscallExec(cmd[0], cmd, envForExec(nil))
 	}
 }
+
+func randomUUID() (string, error) {
+	// The source of the bytes generated here is the same as that of
+	// /dev/urandom which doesn't block and is sufficient for our purposes
+	// of avoiding clashing UUIDs that are needed for all of the non-service
+	// commands that are started with the help of this UUID.
+	uuidBytes, err := ioutil.ReadFile("/proc/sys/kernel/random/uuid")
+	return strings.TrimSpace(string(uuidBytes)), err
+}
+
+var createTransientScope = func(securityTag string) error {
+	if !features.RefreshAppAwareness.IsEnabled() {
+		return nil
+	}
+
+	logger.Debugf("creating transient scope %s", securityTag)
+	// The scope is created with a DBus call to systemd running either on
+	// system or session bus, depending on if we are starting a program as root
+	// or as a regular user.
```

Comment:
I confirmed (on 20.04) that this does *not* work correctly and we definitely need a comment mentioning this limitation. Ie, this is currently not supported:
```
$ id -u other-user
1001
$ id -u
1000
$ sudo [-i] -u other-user test-snapd-sh.sh
```

This is because while sudo started as uid 1000 goes through the pam stack, the systemd user session for uid 1001 (other-user) is not running and nothing in the sudo pam stack will start it.

*If* a user session happens to already be running for uid 1001 (other-user; eg, logged in the desktop/ssh/console session as other-user already), then this invocation of sudo started as uid 1000 does work.